cms/djangoapps/contentstore/views/item.py

```diff
@@ -1135,10 +1134,12 @@ def create_xblock_info(xblock, data=None, metadata=None, include_ancestor_info=F
     if is_concise:
         if child_info and len(child_info.get('children', [])) > 0:
             xblock_info['child_info'] = child_info
+        user_partitions = get_user_partition_info(xblock, schemes=['random'], course=course)
```

Comment:
But user_partitions is a variable that is returned in the xblock_info dict. I think it should be all the user_partitions, not just the random ones. You don't know how callers of create_xblock_info are going to use the returned data, and there is a good chance that they need all the user_partitions.

I would go back to your previous solution, since it means that overriding of the display_name only happens for xblocks related to the random user partitions (since get_group_display_name does the filtering, and you only replace the display_name if group_display_name is not None). And then the user_partitions returned by create_xblock_info will continue to be all defined user_partitions.